kafka-daq-report/src/main.rs

```diff
@@ -238,7 +236,16 @@ async fn poll_kafka_msg(consumer: StreamConsumer, shared_data: SharedData) {
                                         d.num_msg_received += 1;
 
                                         d.last_msg_timestamp = timestamp;
-                                        d.message_rate = message_rate;
+
+                                        d.message_rate = if let (Some(first), Some(last)) =
```

Comment:
Could a message count over a fixed time window be used instead.

This approach would have issues responding to and accurately representing unstable message receive rates, plus reliance on the timestamps should be avoided.

Reply:
Yes, that makes sense. Should the fixed time window be hard-coded in or passed in as a command line argument?

Reply:
A command line argument with a default of 5 seconds would be good.